### PR TITLE
Remove @types/adal in favor of @types/adal-angular

### PR DESCRIPTION
**Pull request by unindented**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If removing a declaration:
- [x] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by unindented** [comment]
See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/22549

File: notNeededPackages.json [hunk 1]
```diff
@@ -6,6 +6,12 @@
             "sourceRepoURL": "https://github.com/betterwaysystems/packer",
             "asOfVersion": "1.1.3"
         },
+        {
+            "libraryName": "Active Directory Authentication Library for JavaScript",
```
Comment (paulvanbrenk) [inline notNeededPackages.json]: This is used when the definition file is part of the actual package, in this case you replaces this definition with a new one. So you don't need this.

**Review by paulvanbrenk — CHANGES_REQUESTED** [review]

File: notNeededPackages.json (hunk 1 above)
Comment (unindented) [reply notNeededPackages.json]: Ah, understood. I've removed my changes to the `notNeededPackages` file.

**Comment by mhegazy** [comment]
@andy-ms do we need to add these packages to notNeededPackages.json?

**Comment by andy-ms** [comment]
@mhegazy There's no `adal` on npm so that wouldn't work. But we can manually deprecate `@types/adal`.

**Comment by mhegazy** [comment]
@andy-ms should i merge this first, or you will deprecate first. we probably need some sort of a formal deprecation process for such packages.

**Comment by mhegazy** [comment]
feel free to merge as well.

**Comment by andy-ms** [comment]
Ran `npm deprecate @types/adal "'@types/adal' is now '@types/adal-angular'"`

**Pull request merged**